src/App.tsx

```diff
@@ -6,54 +6,151 @@ import todosFromServer from './api/todos';
 import { TodoList } from './components/TodoList';
 import { Todo } from './types/Todo';
 import { TodoForm } from './components/TodoForm';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import React, { useState } from 'react';
+
+interface Todo {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  user: User | null;
+}
+
+interface User {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+}
 
-const todoWithUser: Todo[] = todosFromServer.map(todo => ({
+function getUserId(userId: number): User {
+  return usersFromServer.find(user => user.id === userId) as User;
+}
+
+const todos = todosFromServer.map(todo => ({
   ...todo,
-  user: usersFromServer.find(user => user.id === todo.userId) || null,
+  user: getUserId(todo.userId),
 }));
 
 export const App = () => {
-  const [todos, setTodos] = useState<Todo[]>([]);
-
-  const addTodo = (newTodo: Omit<Todo, 'id'>) => {
-    const todoToAdd = {
-      id: Math.max(...todos.map(todo => todo.id)) + 1,
-      ...newTodo,
-    };
-
-    setTodos((currentTodos) => [...currentTodos, todoToAdd]);
-  };
-
-  const deleteTodo = (todoID: number) => {
-    setTodos((currentTodos) => currentTodos.filter(
-      todo => todo.id !== todoID,
-    ));
-  };
-
-  // const updateTodo = (updatedTodo: Todo) => {
-  //   setTodos((currentTodos) => currentTodos.map(
-  //     todo => (todo.id === updatedTodo.id ? updatedTodo : todo),
-  //   ));
-  // };
-
-  useEffect(() => {
-    setTimeout(() => {
-      setTodos(todoWithUser);
-    }, 500);
-  }, []);
+  const [inputText, setinputText] = useState('');
+  const [inputUser, setInputUser] = useState(0);
+  const [isText, setIsText] = useState(true);
+  const [isUser, setIsUser] = useState(true);
+
+  const [listOfTodos, setListOfTodos] = useState(todos);
+
+  function getTheLargestTodoId(todoList: Todo[]) {
+    return [...todoList].sort(
+      (aTodo: Todo, bTodo: Todo) => bTodo.id - aTodo.id,
+    )[0].id;
+  }
```

Comment:
Why not store the largest todoId in a state and just increment it by 1 when creating a new todo?